src/Parsers/ASTCreateFunctionQuery.h

```diff
@@ -16,6 +19,13 @@ class ASTCreateFunctionQuery : public IAST, public ASTQueryWithOnCluster
     bool or_replace = false;
     bool if_not_exists = false;
 
+    /// proton: starts
+    bool is_aggregation = false;
+    bool is_javascript_func = false;
```

Comment:
Shall we use `std::string lang;` (when `lang = JavaScript` is js) for generalization and provide a help `bool isJavasScript() const noexpect` help function

Reply:
What about customized emit strategy support ?

Reply:
I am thinking if we shall generate a structure for all of these data members to make them explicit. Something like 

```
struct UDF
{
   bool is_aggregation = false;
   std::string lang;
   ASTPtr arguments;
   ASTPtr return_type;

   bool isJavaScript() const noexcept { return lang == "JAVASCRIPT"; }
}; 
```